Sources/StreamChatUI/ChatMessageList/ChatMessageListView+DiffKit.swift

```diff
@@ -22,8 +22,7 @@ extension ChatMessageListView {
         CATransaction.setCompletionBlock(completion)
         reload(
             using: changeset,
-            with: animation(),
-            reconfigure: { _ in false }
```

Comment:
Btw, why don't we need this for the Message List ? 🤔

Reply:
I did some testing with message list and did not see noticeable improvement (in time profiler). There are a couple of reasons. 
1. `Stagedchangeset.elementUpdated` does not happen often
2. Message list uses many different reuse identifiers, even when we get `elementUpdated` it is highly probable that the computed reuse identifier changes and in that case we must call reload row (table view throws an exception if we call reconfigure but reuse identifier changes)
3. For actually making this work, this closure needs to compare the current reuse identifier by accessing the current cell's reuseIdentifier and also computing the new reuseIdentifier. This gets a bit complex since this code is in ChatMessageListView but it needs to access some functions from ChatMessageListVC for being able to compute the reuse identifier.
```
let reuseIdentifier = self.reuseIdentifier(
            contentViewClass: contentViewClass, <- function on ChatMessageListVC
            attachmentViewInjectorType: attachmentViewInjectorType, <- function on ChatMessageListVC
            layoutOptions: layoutOptions, <- function on ChatMessageListVC
            message: message <- function on ChatMessageListVC
        )
```

After taking all of this into account it felt like it is actually not worth it but I can be convinced otherwise (I did some hack implementation while I was measuring).

Instruments runs (reconfigure rows did show 2 less hitches :))
<img width="2560" alt="messagelist" src="https://github.com/GetStream/stream-chat-swift/assets/1469907/8a6ddbdc-fe31-4ba0-85f5-4609918a64c1">